### PR TITLE
grub: add extra patches for ieee1275 + disable altivec for ppc64 + font fix

### PR DESCRIPTION
**Pull request by q66**
This makes grub-ieee1275 actually work. Patches are from Ubuntu.

### CODE REVIEW

**Comment by q66** [comment]
Looks like the build break on i686 is for binutils 2.30+ and is not related to my changes. It seems that it might be necessary to rebuild binutils with some additional flags.

**Comment by q66** [comment]
Okay, looks like it will not build binutils with the changes first... anyone mind testing i686 with this?

**Comment by q66** [comment]
This is the issue: https://lists.gnu.org/archive/html/bug-grub/2018-01/msg00006.html

This is the suggested solution: http://archive.linuxfromscratch.org/mail-archives/lfs-support/2018-January/051724.html

File: srcpkgs/grub/template [hunk 1]
```diff
@@ -50,6 +50,13 @@ do_configure() {
 	unset CC AS LD RANLIB CPP
 	CFLAGS="$CFLAGS -fno-stack-protector"
 
+	# building with altivec generates broken grub core
+	case "$XBPS_TARGET_MACHINE" in
+		ppc64*)
+			CFLAGS="$CFLAGS -mno-altivec"
```
Comment (pullmoll) [inline srcpkgs/grub/template]: This could be on one line  `ppc64*) CFLAGS+=" -mno-altivec";;` like in many other places. xbps-src requires bash.

**Comment by q66** [comment]
Hm, no idea how to fix this then - the internet hasn't been too helpful

**Comment by pullmoll** [comment]
I think this has to do with the uncommon i686 triplet `i686-pc-linux-gnu` (pc instead of unknown) which for some reason maps to `x86_64-pc-none` instead of `x86_64-unknown-none` or perhaps `x86_64-unknown-linux`. OTOH the same error happens for `i686-musl` where the triplet is `i686-unknown-linux-musl`.

**Comment by q66** [comment]
Integrated https://github.com/void-linux/void-packages/pull/11942 to not have to bump twice.

**Comment by q66** [comment]
@pullmoll, can you try on i686 now?

**Comment by pullmoll** [comment]
@q66 yes, `binutils` built here for `i686{,-musl}` and so did `grub`.
I think this can be merged now @void-linux/pkg-committers ?

**Comment by q66** [comment]
Need to merge binutils separately, wait for builders, then grub - as binutils is core chroot package, not a dependency.

**Comment by q66** [comment]
Binutils PR here: https://github.com/void-linux/void-packages/pull/11953 We need to merge that first, and then this one can go.

**Comment by pullmoll** [comment]
If we want to install `grub-x86_64-efi` also for the `i686` ISOs, we will have to manually add this noarch package to `i686-repodata` because it isn't built. I have no idea if it makes sense to have a x86_64-efi on a i686 installer ISO, though. Perhaps we should change the `void-mklive` scripts to add only `grub-i386-efi`  for `i686`?

**Comment by q66** [comment]
AFAIK `i386-efi` should be enough. Also, `mklive` assumes an `x86_64` host for now, even for building `i686` ISOs.

**Comment by q66** [comment]
Still getting that CI issue there, old masterdir?

**Comment by pullmoll** [comment]
Yes, Travis pulls in the i686 [docker masterdir 2019042701](https://travis-ci.org/void-linux/void-packages/jobs/537971891#L379) and this can not yet have the binutils-2.32_2 update which I merged to master on 2019-05-26 09:28:09.

**Pull request closed without merging**